lib/tasks/migrate/20230730055732_migrate_from_folders_to_lists.rake

```diff
@@ -2,18 +2,30 @@ namespace :check do
   namespace :migrate do
     task fix_tags_with_invalid_data: :environment do
       started = Time.now.to_i
-      Annotation.where(annotation_type: "tag", annotated_type: "ProjectMedia")
-      .find_in_batches(:batch_size => 1000) do |tags|
-        print '.'
-        tags.each do |tag|
-          tag_text = tag.data['tag']
-          if tag_text.class.name == 'TagText'
+      limit = 500
+      offset = 0
+      errors = []
+      loop do
+        puts "Query tags with limit [#{limit}] and offset [#{offset}]"
+        query = "SELECT id, data FROM annotations WHERE annotation_type = 'tag' ORDER BY id LIMIT $1 OFFSET $2"
```

Comment:
Since this runs while the live database is still online and new records are written, I suggest you query until a `last_id`:

```
last_id = Annotation.where(annotation_type: 'tag').last.id
query = "SELECT id, data FROM annotations WHERE annotation_type = 'tag' AND id < $1 ORDER BY id ASC LIMIT $2 OFFSET $3"
result = ActiveRecord::Base.connection.exec_query(query, 'tag query', [last_id, limit, offset]).to_a
```

Reply:
Done